src/structures/vroom/input/input.cpp

```diff
@@ -602,9 +602,15 @@ void Input::set_vehicles_TSP_flag() {
 
 void Input::set_vehicles_costs() {
   for (auto& vehicle : vehicles) {
-    auto d_m = _durations_matrices.find(vehicle.profile);
-    assert(d_m != _durations_matrices.end());
-    vehicle.cost_wrapper.set_durations_matrix(&(d_m->second));
+    auto duration_m = _durations_matrices.find(vehicle.profile);
+    assert(duration_m != _durations_matrices.end());
+    vehicle.cost_wrapper.set_durations_matrix(&(duration_m->second));
+
+    // TODO this will probably break with user-provided durations but
+    // not distances.
+    auto distance_m = _distances_matrices.find(vehicle.profile);
+    assert(distance_m != _distances_matrices.end());
+    vehicle.cost_wrapper.set_distances_matrix(&(distance_m->second));
```

Comment:
This indeed causes the CI build to break on the usual check with our `example_2.json` file because no distances matrix is provided along with the durations matrix. To shadow this problem, we could simply update example 2 with the distances matrix (it makes sense to do so anyway).

But we should decide on how to retro-compatibly handle the situation of custom matrices with durations (maybe costs) but no distances. I think the best default way of doing it would be to create a distances matrices filled with zeros. This way it would not impose any change on costs and constraints.